Lib/test/test_android.py

```
@@ -153,6 +161,22 @@ def write(s, lines=None):
                 write("hello\r\nworld\r\n", ["hello", "world"])
                 write("\r\n", [""])
 
+                # Non-standard line separators should be preserved.
+                write("before form feed\x0cafter form feed\n",
+                      ["before form feed\x0cafter form feed"])
+                write("before line separator\u2028after line separator\n",
+                      ["before line separator\u2028after line separator"])
+
+                # String subclasses are accepted, and if their methods write
+                # themselves, this doesn't cause infinite recursion.
+                class CustomStr(str):
+                    def splitlines(self, *args, **kwargs):
+                        sys.stdout.write(self)
+                        return super().splitlines(*args, **kwargs)
```

You can just return an arbitrary result or raise an arbitrary error. This method should never be called. You can also override `__len__` and `__str__`. The latter would catch a bug if you use `s = str(s)` instead of `s = str.__str__(s)` to convert to a string.

---

Done.